File: app/src/main/java/com/candy/capture/core/CaptureApplication.java
package com.candy.capture.core;

import android.app.Application;
import android.content.Intent;

import com.candy.capture.BuildConfig;
import com.candy.capture.service.FloatingWindowService;
import com.candy.capture.util.FileUtil;
import com.candy.commonlibrary.utils.LogUtil;
import com.umeng.analytics.MobclickAgent;

/**
 * Created by zhanghq on 2016/10/19.
 */

public class CaptureApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        FileUtil.initRootFolder(this);
        if (!BuildConfig.DEBUG) {
            MobclickAgent.setDebugMode(true);
            LogUtil.disableLog();
        }

//        Intent intent = new Intent(this, FloatingWindowService.class);
//        startService(intent);
    }
}

File: app/src/main/java/com/candy/capture/core/CaptureApplication.kt
package com.candy.capture.core

import android.app.Application
import android.content.Context
import com.candy.capture.BuildConfig
import com.candy.capture.util.FileUtil
import com.candy.commonlibrary.utils.LogUtil
import com.umeng.analytics.MobclickAgent

/**
 * @Description
 * @Author zhanghaiqiang
 * @Date 2020/9/30 11:19
 */
class CaptureApplication: Application() {
    override fun onCreate() {
        super.onCreate()
        instance = this
        FileUtil.initRootFolder(this)
        if (!BuildConfig.DEBUG) {
            MobclickAgent.setDebugMode(true)
            LogUtil.disableLog()
        }
    }

    companion object {
        private var instance: Context? = null
        fun getInstance(): Context {
            return instance!!
        }
    }
}